crates/cast/src/main.rs

```diff
@@ -372,3 +410,59 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
 
     config.accounts_file = Utf8PathBuf::from(shellexpand::tilde(&new_accounts_file).to_string());
 }
+
+struct PackageData {
+    pub manifest_path: Result<Utf8PathBuf>,
+    pub metadata: Result<Metadata>,
+    pub package: Result<PackageMetadata>,
+}
+
+// Lazily get all the required package information from CLI
+fn get_package_data(cli: &Cli) -> PackageData {
+    let mut result = PackageData {
```

Comment:
The problem if it fails for example on metadata step then we have no real way to get the package metadata, those early returns help to avoid nested matching

Reply:
:c kinda right